homeassistant/components/fritzbox/light.py

```diff
@@ -0,0 +1,168 @@
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the FRITZ!SmartHome light from ConfigEntry."""
+    entities: list[FritzboxLight] = []
+    coordinator = hass.data[FRITZBOX_DOMAIN][entry.entry_id][CONF_COORDINATOR]
+
+    for ain, device in coordinator.data.items():
+        if not device.has_lightbulb:
+            continue
+
+        supported_color_temps = await hass.async_add_executor_job(
+            device.get_color_temps
+        )
+
+        supported_colors = await hass.async_add_executor_job(device.get_colors)
+
+        entities.append(
+            FritzboxLight(
+                {
+                    ATTR_NAME: f"{device.name}",
+                    ATTR_ENTITY_ID: f"{device.ain}",
+                    ATTR_UNIT_OF_MEASUREMENT: None,
+                    ATTR_DEVICE_CLASS: None,
+                    ATTR_STATE_CLASS: None,
+                },
+                coordinator,
+                ain,
+                supported_colors,
+                supported_color_temps,
+            )
+        )
+
+    async_add_entities(entities)
+
+
+class FritzboxLight(FritzBoxEntity, LightEntity):
+    """The light class for FRITZ!SmartHome lightbulbs."""
+
+    def __init__(
+        self,
+        entity_info: EntityInfo,
+        coordinator: DataUpdateCoordinator[dict[str, FritzhomeDevice]],
+        ain: str,
+        supported_colors: dict,
+        supported_color_temps: list[str],
+    ) -> None:
+        """Initialize the FritzboxLight entity."""
+        super().__init__(entity_info, coordinator, ain)
+
+        max_kelvin = int(max(supported_color_temps))
+        min_kelvin = int(min(supported_color_temps))
+
+        # max kelvin is min mireds and min kelvin is max mireds
+        self._attr_min_mireds = color.color_temperature_kelvin_to_mired(max_kelvin)
+        self._attr_max_mireds = color.color_temperature_kelvin_to_mired(min_kelvin)
+
+        # Fritz!DECT 500 only supports 12 values for hue, with 3 saturations each.
+        # Map supported colors to dict {hue: [sat1, sat2, sat3]} for easier lookup
+        self._supported_hs = {}
+        for values in supported_colors.values():
+            hue = int(values[0][0])
+            self._supported_hs[hue] = [
+                int(values[0][1]),
+                int(values[1][1]),
+                int(values[2][1]),
+            ]
+
+    @property
+    def is_on(self) -> bool:
+        """If the light is currently on or off."""
+        return self.device.state  # type: ignore [no-any-return]
+
+    @property
+    def brightness(self) -> int:
+        """Return the current Brightness."""
+        return self.device.level  # type: ignore [no-any-return]
+
+    @property
+    def hs_color(self) -> tuple[float, float] | None:
+        """Return the hs color value."""
+        # Don't return hue and saturation unless in color mode
+        if self.device.color_mode != "1":
```

Comment:
```suggestion
        if self.device.color_mode != COLOR_MODE:
```

define `COLOR_MODE` as constant, those comment can be removed, since code is self explaining